packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.classNames.ts

```diff
@@ -1,28 +1,125 @@
 import { memoizeFunction } from '../../Utilities';
-import { mergeStyleSets, ITheme } from '../../Styling';
-import { IContextualMenuStyles } from './ContextualMenu.Props';
-
+import { ITheme, mergeStyleSets } from '../../Styling';
+import { IContextualMenuStyles, IMenuItemStyles } from './ContextualMenu.Props';
+import { getStyles as getContextualMenuStyles, getMenuItemStyles } from './ContextualMenu.styles';
 export interface IContextualMenuClassNames {
-  title?: string;
+  container: string;
+  root: string;
+  list: string;
+  divider: string;
+  header: string;
+  title: string;
+}
+
+export interface IMenuItemClassNames {
+  item: string;
+  root: string;
+  linkContent: string;
+  icon: string;
+  subMenuIcon: string;
+  label: string;
 }
 
-export const getClassNames = memoizeFunction((
-  styles: IContextualMenuStyles,
+export const getContextualMenuClassNames = memoizeFunction((
   theme: ITheme,
+  className: string
 ): IContextualMenuClassNames => {
 
-  styles = styles || {};
+  const styles = getContextualMenuStyles(theme);
 
   return mergeStyleSets({
-    title: [{
-      fontSize: '16px',
-      paddingRight: '14px',
-      paddingLeft: '14px',
-      paddingBottom: '5px',
-      paddingTop: '5px',
-      backgroundColor: theme.palette.neutralLight
-    },
-    styles.title
-    ]
+    container: [
+      'ms-ContextualMenu-container',
+      className,
```

Comment:
So I think we want className to go after the styles.*. If a consumer provides a className that came from merge styles, they'd want the styles from that className to be applied on top of the default styles.

Reply:
Agree... let me update it for the other classnames too

Reply:
@christiango I see this problem... in other files too... should we open an issue for them?

Reply:
Yeah, we should.